### 1. Introduction

### 2. Importing the Libraries

In [1]:
import  numpy as np
import pandas as pd

import matplotlib.pyplot as plt
import seaborn as sns

import warnings
warnings.filterwarnings('ignore')

### 3. Loading the Dataset

In [2]:
df = pd.read_csv('sneakers.csv')

In [3]:
sneakers = df.copy()
sneakers.head()

Product Name Product ID  \
0  Women's adidas Originals NMD_Racer Primeknit S...     AH2430   
1               Women's adidas Originals Sleek Shoes     G27341   
2                  Women's adidas Swim Puka Slippers     CM0081   
3   Women's adidas Sport Inspired Questar Ride Shoes     B44832   
4           Women's adidas Originals Taekwondo Shoes     D98205   

   Listing Price  Sale Price  Discount                    Brand  Rating  \
0          14999        7499        50  Adidas Adidas ORIGINALS     4.8   
1           7599        3799        50         Adidas ORIGINALS     3.3   
2            999         599        40        Adidas CORE / NEO     2.6   
3           6999        3499        50        Adidas CORE / NEO     4.1   
4           7999        3999        50         Adidas ORIGINALS     3.5   

   Reviews  
0       41  
1       24  
2       37  
3       35  
4       72

In [4]:
print('the shape of the DataFrame is: ', sneakers.shape)
print('the size of the DataFrame is: ', sneakers.size)
print('the dimension of the DataFrame is: ', sneakers.ndim)
print('the column names of the DataFrame are:\n ', sneakers.columns)
print('the dtypes of the DataFrame is:\n ', sneakers.dtypes)

the shape of the DataFrame is:  (3268, 8)
the size of the DataFrame is:  26144
the dimension of the DataFrame is:  2
the column names of the DataFrame are:
  Index(['Product Name', 'Product ID', 'Listing Price', 'Sale Price', 'Discount',
       'Brand', 'Rating', 'Reviews'],
      dtype='object')
the dtypes of the DataFrame is:
  Product Name      object
Product ID        object
Listing Price      int64
Sale Price         int64
Discount           int64
Brand             object
Rating           float64
Reviews            int64
dtype: object


In [5]:
sneakers.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3268 entries, 0 to 3267
Data columns (total 8 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Product Name   3268 non-null   object 
 1   Product ID     3268 non-null   object 
 2   Listing Price  3268 non-null   int64  
 3   Sale Price     3268 non-null   int64  
 4   Discount       3268 non-null   int64  
 5   Brand          3268 non-null   object 
 6   Rating         3268 non-null   float64
 7   Reviews        3268 non-null   int64  
dtypes: float64(1), int64(4), object(3)
memory usage: 204.4+ KB


In [6]:
sneakers.describe()

Listing Price    Sale Price     Discount       Rating      Reviews
count    3268.000000   3268.000000  3268.000000  3268.000000  3268.000000
mean     6868.020196   6134.265606    26.875765     3.242105    40.551714
std      4724.659386   4293.247581    22.633487     1.428856    31.543491
min         0.000000    449.000000     0.000000     0.000000     0.000000
25%      4299.000000   2999.000000     0.000000     2.600000    10.000000
50%      5999.000000   4799.000000    40.000000     3.500000    37.000000
75%      8999.000000   7995.000000    50.000000     4.400000    68.000000
max     29999.000000  36500.000000    60.000000     5.000000   223.000000

In [7]:
sneakers.describe(include='object')

Product Name  Product ID  \
count                                       3268        3268   
unique                                      1531        3179   
top     Women's adidas Originals Sambarose Shoes  CD6720-808   
freq                                          16           3   

                    Brand  
count                3268  
unique                  5  
top     Adidas CORE / NEO  
freq                 1111

In [8]:
sneakers.isnull().sum()

Product Name     0
Product ID       0
Listing Price    0
Sale Price       0
Discount         0
Brand            0
Rating           0
Reviews          0
dtype: int64

In [9]:
sneakers[sneakers.duplicated()].count()

Product Name     88
Product ID       88
Listing Price    88
Sale Price       88
Discount         88
Brand            88
Rating           88
Reviews          88
dtype: int64

In [10]:
sneakers.drop_duplicates(inplace=True, ignore_index=True)

In [11]:
sneakers[sneakers.duplicated()].count()

Product Name     0
Product ID       0
Listing Price    0
Sale Price       0
Discount         0
Brand            0
Rating           0
Reviews          0
dtype: int64

In [12]:
sneakers.shape

(3180, 8)

In [13]:
highest = sneakers[['Product Name', 'Rating', 'Reviews']]
highest.head()

Product Name  Rating  Reviews
0  Women's adidas Originals NMD_Racer Primeknit S...     4.8       41
1               Women's adidas Originals Sleek Shoes     3.3       24
2                  Women's adidas Swim Puka Slippers     2.6       37
3   Women's adidas Sport Inspired Questar Ride Shoes     4.1       35
4           Women's adidas Originals Taekwondo Shoes     3.5       72

In [14]:
highest.sort_values(by='Rating', ascending=False).head(10)

Product Name  Rating  Reviews
1039     Men's adidas Sport Inspired Court Adapt Shoes     5.0       96
2964                     Nike React Element 55 Premium     5.0        1
2488  Men's adidas Originals Stan Smith Gore-Tex Shoes     5.0       32
2954                            Nike Air Max 720 Waves     5.0        4
2953                         Nike Air Zoom Wildhorse 5     5.0        6
1487    Men's adidas Sport Inspired VL Court 2.0 Shoes     5.0       33
2949                              Nike Air Streak Lite     5.0        1
221             Men's adidas Running Stargon 1.0 Shoes     5.0       61
595                   Men's Running Adispree 5.0 Shoes     5.0       65
2945                                    Nike Zoom KD12     5.0        1

In [15]:
highest.sort_values(by='Reviews', ascending=False).head(10)

Product Name  Rating  Reviews
2750                                Air Jordan 10 Retro     4.7      223
2804                                      Nike Zoom Fly     4.3      105
2166          Men's adidas Originals Forest Grove Shoes     4.5       99
723   Women's adidas Sport Inspired Farm Rio Advanta...     4.6       99
667            Men's adidas RUNNING Stardrift Low Shoes     2.9       99
1568                    Men's Originals Drop Step Shoes     1.0       99
89                  Men's adidas Running Galaxy 4 Shoes     2.4       99
1149          Men's adidas Originals A.R. Trainer Shoes     4.0       99
560               Men's adidas Outdoor Geocach 19 Shoes     2.4       99
1274                      Men's Running Nova Flow Shoes     3.1       99

In [16]:
sneakers.sort_values(by='Reviews', ascending=False).head()

Product Name  Product ID  \
2750                                Air Jordan 10 Retro  310805-137   
2804                                      Nike Zoom Fly  880848-005   
2166          Men's adidas Originals Forest Grove Shoes      CG5675   
723   Women's adidas Sport Inspired Farm Rio Advanta...      EF0133   
667            Men's adidas RUNNING Stardrift Low Shoes      CI1741   

      Listing Price  Sale Price  Discount              Brand  Rating  Reviews  
2750              0       15995         0               Nike     4.7      223  
2804              0       14495         0               Nike     4.3      105  
2166           6999        4199        40   Adidas ORIGINALS     4.5       99  
723            5999        3599        40  Adidas CORE / NEO     4.6       99  
667            4999        2000        60  Adidas CORE / NEO     2.9       99

In [17]:
# 6.
sneakers['Brand'].value_counts()

Brand
Adidas CORE / NEO           1111
Adidas ORIGINALS             907
Adidas SPORT PERFORMANCE     606
Nike                         555
Adidas Adidas ORIGINALS        1
Name: count, dtype: int64

In [18]:
# 7.
sneakers[sneakers['Listing Price'] == 0].value_counts().sum()

362

In [19]:
# 8.


In [20]:
# 9.

In [21]:
# 13.
sneakers.groupby('Brand')['Discount'].value_counts()

Brand                     Discount
Adidas Adidas ORIGINALS   50            1
Adidas CORE / NEO         50          521
                          40          269
                          0           212
                          20           44
                          30           42
                          60           23
Adidas ORIGINALS          50          298
                          40          291
                          0           282
                          60           20
                          30           16
Adidas SPORT PERFORMANCE  40          203
                          50          174
                          0           165
                          30           41
                          60           18
                          20            5
Nike                      0           555
Name: count, dtype: int64

In [22]:
sneakers.groupby('Brand')['Discount'].count()

Brand
Adidas Adidas ORIGINALS        1
Adidas CORE / NEO           1111
Adidas ORIGINALS             907
Adidas SPORT PERFORMANCE     606
Nike                         555
Name: Discount, dtype: int64

- **Answer all questions here (after each question).**

1. What is the shape of the dataset? The Shape of the dataset is (3268, 8).
2. Are there any missing values in this dataset? There are **no** missing values in this dataset.
3. List all 8 columns' dtypes.
     - Product Name -- object
     - Product ID -- object
     - Listing Price -- int64
     - Sale Price -- int64
     - Discount -- int64
     - Brand -- object
     - Rating -- float64
     - Reviews -- int64
4. Are there any duplicated rows (observations) in this dataset? There are 88 rows of duplicates in this dataset.
5. Refer to the statistical summary of the dataset and answer the following:
     - The average listing price: 6868
     - The average sale price: 6134.2
     - The average discount: 22.6
     - The average rating: 3.2
     - The average number of reviews: 40.5
6. Which "Brand" has the highest number of products in this dataset? <code>Adidas CORE / NEO           1111</code>
7. As you can see, the "Listing Price" for some products is zero (0). How many products with zero listing price are there in this dataset? <code>362</code>
8. Perform the "Univariate EDA", and examine any type of skewness (outliers) for all the variables.
9. The the percentage for the following:
     - Products with listing prices less than 15000
     - Products with sale price less than 15000
     - Products with at least 50% discount
     - Products with rating less than 3.5
     - Products with review numbers less than 100
     - The 3 highest percentages of the discount graph
10. Perform the "Bivariate EDA", and examine the following:
     - The two highly positively correlated variables: <code>'Sale Price' and 'Listing Price'</code>
     - The two highly negatively correlated variables: <code>'Discount' and 'Sale Price'</code>
11. What are the names of the 10 highest rated products?
12. What are the 10 products' names with the highest number of reviews?
13. Which company gives more discount: Adidas or Nike?